turbo/engineapi/engine_server.go

```diff
@@ -944,3 +958,18 @@ func (e *EngineServer) HandlesForkChoice(
 	}
 	return payloadStatus, nil
 }
+
+func waitForStuff(waitCondnF func() (bool, error)) (bool, error) {
+	shouldWait, err := waitCondnF()
+	if err != nil || !shouldWait {
+		return false, err
+	}
+	for i := 0; i < 1000; i++ {
```

Comment:
The wait should be capped at 1s per the [engine API spec](https://github.com/ethereum/execution-apis/blob/main/src/engine/paris.md) (check for timeouts). Please reduce the time to 1s and put a comment referring to the spec timeout.

Reply:
Reduced to 8 seconds.
Timeout from the spec's perspective, in this case, is a loose idea (it's not a protocol parameter). Post-merge chain operations like these are, and should be, driven by the CL. CL should ideally "move on" after a certain timeout lest it misses the next slot. EL should implement a sane "timeout" so as to not get stuck in a loop or something.
8 sec timeout since, this func is only being used in FCU and newPayload, and is based on Ethereum's spec while we do support other chains.
The change in this PR has been done as Erigon would decide, within milliseconds, that it's not ready and is "SYNCING" - something not widely understood or implemented as a part of the protocol. In my opinion, we should have the longest possible timeout, even stretching beyond the slot time, as a CL may decide to issue FCU and newPayload in advance (for whatever reason, again CL is the driver)